input/dir/dir_input.go

```diff
@@ -0,0 +1,55 @@
+package dir
+
+import (
+	"crypto/sha512"
+	"hash"
+	"os"
+
+	"github.com/spacemonkeygo/errors"
+	"polydawn.net/repeatr/def"
+	"polydawn.net/repeatr/input"
+	"polydawn.net/repeatr/lib/fshash"
+)
+
+const Type = "dir"
+
+// interface assertion
+var _ input.Input = &Input{}
+
+type Input struct {
+	spec          def.Input
+	hasherFactory func() hash.Hash
+}
+
+func New(spec def.Input) *Input {
+	if spec.Type != Type {
+		panic(errors.ProgrammerError.New("This input implementation supports definitions of type %q, not %q", Type, spec.Type))
```

Comment:
Still feels like belt & suspenders to me, but it may as well stay for now. Conversation will likely make more sense in light of a executor PR that I'll file later today.